test/gateway/ost_prime/unwrap.js

```diff
@@ -0,0 +1,192 @@
+
+    let brandedTokenAddress, ostPrime, callerAddress, amount;
+
+    async function initialize(){
+      await ostPrime.initialize(
+        {from: accounts[2], value:TOKENS_MAX}
+      );
+    };
+
+    beforeEach(async function () {
+
+      brandedTokenAddress = accounts[2];
+      ostPrime = await OSTPrime.new(brandedTokenAddress);
+
+      callerAddress = accounts[3];
+      amount = new BN(500);
+
+      await ostPrime.setTokenBalance(callerAddress, amount);
+
+    });
+
+    it('Should fail when the amount is zero', async function () {
+      await initialize();
+
+      let amount = 0;
+      await Utils.expectRevert(
+        ostPrime.unwrap.call(amount, { from: callerAddress }),
+        "Amount should not be zero."
+      );
+
+    });
+
+    it('Should fail when contract is not initialized', async function () {
+
+      await Utils.expectRevert(
+        ostPrime.unwrap.call(amount, { from: callerAddress }),
+        "Contract is not initialized."
+      );
+
+    });
+
+    it('Should fail when amount is greater than the account balance',
+      async function () {
+
+      await initialize();
+
+      let amount = new BN(501);
+      await Utils.expectRevert(
+        ostPrime.unwrap.call(amount, { from: callerAddress }),
+        "Insufficient balance."
+      );
+
+    });
+
+    it('Should pass with correct parameters', async function () {
+      await initialize();
+
+      let initialContractBalance = new BN(
+        await Utils.getBalance(ostPrime.address)
+      );
+
+      let initialCallerBalance = new BN(
+        await Utils.getBalance(callerAddress)
+      );
+
+      let result = await ostPrime.unwrap.call(amount, { from: callerAddress });
+      assert.strictEqual(
+        result,
+        true,
+        `The contract should return true.`
+      );
+
+      let tx = await ostPrime.unwrap(amount, { from: callerAddress });
+      let gasUsed = new BN(tx.receipt.gasUsed);
+
+      let callerEIP20Tokenbalance = await ostPrime.balanceOf.call(callerAddress);
+      assert.strictEqual(
+        callerEIP20Tokenbalance.eq(new BN(0)),
+        true,
+        `The balance of ${callerAddress} should be zero.`
+      );
+
+      let contractEIP20Tokenbalance = await ostPrime.balanceOf.call(ostPrime.address);
+      assert.strictEqual(
+        contractEIP20Tokenbalance.eq(amount),
+        true,
+        `The balance of OST prime contract should increase by ${amount}.`
+      );
+
+      let finalContractBalance = new BN(
+        await Utils.getBalance(ostPrime.address)
+      );
+
+      let finalCallerBalance = new BN(
+        await Utils.getBalance(callerAddress)
+      );
+
+      assert.strictEqual(
+        finalContractBalance.eq(initialContractBalance.sub(amount)),
+        true,
+        `Contract base token balance should decrease by ${amount}`
+      );
+
+      assert.strictEqual(
+        finalCallerBalance.eq(initialCallerBalance.add(amount).sub(gasUsed)),
+        true,
```

Comment:
More `BN` asserts throughout all test files

Reply:
`initialCallerBalance` is going to be a big number and I need to perform arithmetic operation. 🤔